app/policies/run_policy.rb

```diff
@@ -40,4 +40,9 @@ def user_admin?
   def user_coach?
     user && user.coach?
   end
+
+  def problem_is_visible?
```

Comment:
A better name for this will be `problem_visible?`. There is no need to have `is` in it.